### ISSUE TITLE
Can't start SERVER mode (neither FULL nor SIMPLE)

### ISSUE THREAD

**Issue by cinnamonmatexfce** [optyfr/JRomManager#29]
Am I doing something wrong?
...
</br>
# FULL-mode
```$ cd /mnt/some/path/to/```
[_see https://www.rosehosting.com/blog/how-to-generate-a-self-signed-ssl-certificate-on-linux/_]
```$ cd /mnt/some/path/to/ && mkdir jrommanager_server_dir```
```$ ls /mnt/some/path/to/```
```
test_cert_jrommanager.pfx ###certificate file
jrommanager_server_dir ###directory
```
```$ cd /usr/share/java/jrommanager-bin && ls```
```
JRomManager.jar
JRomManager.sh
JRomManager-multi.sh
JUpdater.jar
lib
webclient
wrapper
```
```$ LANG=en_US.UTF-8 java -cp JRomManager.jar jrm.fullserver.FullServer --cert="/mnt/some/path/to/test_cert_jrommanager.pfx" --client="./webclient" --workpath="/mnt/some/path/to/jrommanager_server_dir"```
```
Error: Could not find or load main class jrm.fullserver.FullServer
Caused by: java.lang.ClassNotFoundException: jrm.fullserver.FullServer
```
</br>

# SIMPLE-mode
(*In ArchLinux the path* `/usr/share/java/` *is owned by* `root:root`)

```
$ LANG=en_US.UTF-8 java -cp JRomManager.jar jrm.server.Server --port=8080 --client=webclient --workpath=.
Jun 07, 2020 12:30:34 PM jrm.misc.Log err
SEVERE: /usr/share/java/jrommanager-bin/./logs
java.nio.file.AccessDeniedException: /usr/share/java/jrommanager-bin/./logs
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:90)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
        at java.base/sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:389)
        at java.base/java.nio.file.Files.createDirectory(Files.java:694)
        at java.base/java.nio.file.Files.createAndCheckIsDirectory(Files.java:801)
        at java.base/java.nio.file.Files.createDirectories(Files.java:787)
        at jrm.server.Server.getLogPath(Server.java:155)
        at jrm.server.Server.main(Server.java:120)
```

**Comment by optyfr** [comment]
Full Server mode is available since release 2.4 (released today and available before if compiled from github master)
Simple server : use an absolute workpath where you have access rights, or use sudo

**Comment by optyfr** [comment]
2.4.0 is now also available for archlinux
I'll think about modifying the wiki for the workpath thing...

**Comment by cinnamonmatexfce** [comment]
You're fast @optyfr ! ;)
Just upgraded right now. Thanks!!!

**Comment by cinnamonmatexfce** [comment]
# SIMPLE-mode
`$ cd /usr/share/java/jrommanager-bin`
`$ java -cp JRomManager.jar jrm.server.Server --port=8080 --client=webclient --workpath="/mnt/path/with/user/permissions/jrommanager_simpleserver"`
or
`$ java -cp JRomManager.jar jrm.server.Server --port=8080 --client="./webclient" --workpath="/mnt/path/with/user/permissions/jrommanager_simpleserver"`
</br>
JRomManager starts server (I can see it creates the folder `jrommanager_simpleserver`) but poiting the browser to`http://192.my.local.IP:8080` returns error `404`.
Both `Server.0.log` and `Server.0.log.lck` are empty.
</br>
</br>
# FULL-mode
`$ cd /usr/share/java/jrommanager-bin`
`$ java -cp JRomManager.jar jrm.fullserver.FullServer --cert="/mnt/path/with/user/permissions/test_cert_jrommanager.pfx" --client="./webclient" --workpath="/mnt/path/with/user/permissions/jrommanager_fullserver"`
```
INFO::main: Logging initialized @238ms to org.eclipse.jetty.util.log.StdErrLog
Opening jdbc:h2: [...mypath...]/jrommanager_fullserver/Server.sys;MV_STORE=FALSE;MODE=MYSQL
INFO:oejs.Server:main: jetty-9.4.29.v20200521; built: 2020-05-21T17:20:40.598Z; git: 77c232aed8a45c818fd27232278d9f95a021095e; jvm 14.0.1+7
INFO:oejs.session:main: DefaultSessionIdManager workerName=node0
INFO:oejs.session:main: No SessionScavenger set, using defaults
INFO:oejs.session:main: node0 Scavenging every 660000ms
INFO:oejsh.ContextHandler:main: Started o.e.j.s.ServletContextHandler@3b96c42e{/,file:///usr/share/java/jrommanager-bin/webclient/,AVAILABLE}
INFO:oejs.AbstractConnector:main: Started HTTP@6f8e8894{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}
INFO:oejus.SslContextFactory:main: x509=X509@497486b3(1,h=[192.my.local.IP],w=[]) for Server@63f8276e[provider=null,keyStore=file:///mnt/path/with/user/permissions/test_cert_jrommanager.pfx,trustStore=null]
INFO:oejs.AbstractConnector:main: Started HTTP2@75d2da2d{SSL, (ssl, alpn, h2, http/1.1)}{0.0.0.0:8443}
INFO:oejs.Server:main: Started @2304ms
```
Poiting the browser to `http://192.my.local.IP:8080` or `http://192.my.local.IP:8443` asks me user and password `HTTP ERROR 401 Unauthorized`.
`myuser` and `mypassword` doesn't work, neither `root` and `rootpassword`.
Both `Server.0.log` and `Server.0.log.lck` are empty.

**Comment by optyfr** [comment]
for --client if set to relative, then it should be relative to current dir, otherwise you should give also an absolute path that point to the webclient folder
for login, use username "admin" and password "admin"

**Comment by cinnamonmatexfce** [comment]
> for --client if set to relative, then it should be relative to current dir, otherwise you should give also an absolute path that point to the webclient folder
>
Ok. Btw, SIMPLE-mode can't start anyway.
```
fi.iki.elonen.NanoHTTPD$Response send
SEVERE: Could not send response to the client
java.net.SocketException: Pipe interrotta
        at java.base/sun.nio.ch.NioSocketImpl.implWrite(NioSocketImpl.java:420)
        at java.base/sun.nio.ch.NioSocketImpl.write(NioSocketImpl.java:440)
        at java.base/sun.nio.ch.NioSocketImpl$2.write(NioSocketImpl.java:826)
        at java.base/java.net.Socket$SocketOutputStream.write(Socket.java:1052)
        at fi.iki.elonen.NanoHTTPD$Response.sendBody(NanoHTTPD.java:1694)
        at fi.iki.elonen.NanoHTTPD$Response.sendBodyWithCorrectEncoding(NanoHTTPD.java:1667)
        at fi.iki.elonen.NanoHTTPD$Response.sendBodyWithCorrectTransferAndEncoding(NanoHTTPD.java:1657)
        at fi.iki.elonen.NanoHTTPD$Response.send(NanoHTTPD.java:1624)
        at fi.iki.elonen.NanoHTTPD$HTTPSession.execute(NanoHTTPD.java:980)
        at fi.iki.elonen.NanoHTTPD$ClientHandler.run(NanoHTTPD.java:192)
        at java.base/java.lang.Thread.run(Thread.java:832)
```
</br>

> for login, use username "admin" and password "admin"
>

Can't login (using port `8443`) ---> `HTTP ERROR 403 Forbidden`
```
URI: | /
-- | --
403
Forbidden
static
```
And if I use port `8080` I get: `SSL_ERROR_RX_RECORD_TOO_LONG`
But I suspect this issue could be related to my dummy `test_cert_jrommanager.pfx`.

**Comment by optyfr** [comment]
simple server : nanohttpd seems broken, maybe java 13 incompatible
full server : 
- 8443 HTTP ERROR 403, hum I think about wrong --client path (or may I forgot something while packaging?)
- 8080 SSL_ERROR_RX_RECORD_TOO_LONG you should not get any SSL error unless you used HTTPS instead of HTTP on your browser... 8080 is dedicated for HTTP

I think I will test on arch myself but  not today

**Comment by cinnamonmatexfce** [comment]
> simple server : nanohttpd seems broken, maybe java 13 incompatible
> 
Let me try previous Java version on my other test machine (always ArchLinux)

**Comment by optyfr** [comment]
Ok, but I'm seriously beginning to wonder if package were complete after gradle finished build...

**Comment by cinnamonmatexfce** [comment]
> > simple server : nanohttpd seems broken, maybe java 13 incompatible
> 
> Let me try previous Java version on my other test machine (always ArchLinux)
>
I installed JRomManager from source (not `-bin`) [https://aur.archlinux.org/packages/jrommanager/](url) with **only Java8** (both for installation and running)... Still same problem: SIMPLE-server starts, but the problem remains:
> JRomManager starts server (I can see it creates the folder jrommanager_simpleserver) but poiting the browser to http://192.my.local.IP:8080 returns error 404.
> Both Server.0.log and Server.0.log.lck are empty.

**Comment by optyfr** [comment]
OK, got it, wrong client packaging, will have to release 2.4.1

**Comment by optyfr** [comment]
Fixed with commit 1f0571d04bd6f0fad7ebdb8e039014cb98493201
